packages/playground/__tests__/context/messages/reducer.test.tsx

```diff
@@ -0,0 +1,89 @@
+import { reducer, defaultState, Action } from '~/context/messages/reducer';
+
+describe('Given the reducer is used to manage state', () => {
+  describe('When "LOG_SYSTEM" action is dispatched', () => {
+    test('When message of type "IN_PROGRESS" is dispatched', () => {
+      const status = 'IN_PROGRESS';
+      const content = 'System message content';
+      const payload: Action = {
+        type: 'LOG_SYSTEM',
+        payload: {
+          content: content,
+          status: status,
+        },
+      };
+      const state = reducer(defaultState, payload);
+
+      // Then ...
+      expect(state.messages[0].id).toBe(0);
+      expect(state.messages[0].prompt).toBe('SYSTEM');
+      expect(state.messages[0].status).toBe(status);
+      expect(state.messages[0].content).toBe(content);
+      expect(state.messages[0].severity).toBe('Warning');
+      expect(state.nextId).toEqual(1);
+    });
+
+    test('When message of type "DONE" is dispatched', () => {
+      const status = 'DONE';
+      const content = 'System message content';
+      const payload: Action = {
+        type: 'LOG_SYSTEM',
+        payload: {
+          content: content,
+          status: status,
+        },
+      };
+      const state = reducer(defaultState, payload);
+
+      // Then ...
+      expect(state.messages[0].id).toBe(0);
+      expect(state.messages[0].prompt).toBe('SYSTEM');
+      expect(state.messages[0].status).toBe(status);
+      expect(state.messages[0].content).toBe(content);
+      expect(state.messages[0].severity).toBe('Success');
+      expect(state.nextId).toEqual(0);
+    });
+
+    test('When message of type "INFO" is dispatched', () => {
+      const status = 'INFO';
+      const content = 'System message content';
+      const payload: Action = {
+        type: 'LOG_SYSTEM',
+        payload: {
+          content: content,
```

Comment:
We can write:

```
const payload: Action = {
        type: 'LOG_SYSTEM',
        payload: {
          content,
          status,
        },
      };
```

here and whenever object keys are identical with their values.